src/index.js

```diff
@@ -63,12 +67,10 @@ const mapProperties = (schema, mapProps) => {
   }
 
   let schemaCopy
-  if (typeof mapProps === 'object') {
-    for (const prop in mapProps) {
-      schemaCopy = schema.map(el => {
-        return replacePropInElement(el, prop, mapProps[prop])
-      })
-    }
+  for (const prop in mapProps) {
```

Comment:
Truth be told, all those for in loops are extremely hard for me to follow. Could we break it up into `.map` and named map functions? I think this should make it easier to follow and understand later on.